### PR TITLE
Fxa 2215 use setup intent api

### LINKED ISSUE #5832: Use auth-server SetupIntent API when updating card information
To properly handle 3D Secure and SCA, we need to use the SetupIntent API with off_session set to true. This will properly trigger authentication and give us the required permissions to charge a card later. The client should be setup to request a new SetupIntent from the server, then use confirmCard which will confirm and add it to the customer on success.

See:

https://stripe.com/docs/billing/migration/strong-customer-authentication#scenario-2

https://stripe.com/docs/payments/save-and-reuse#web-create-setup-intent

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-2215)
┆Issue Number: FXA-2215

### PR DESCRIPTION
**Pull request by jackiemunroe**
## Because:
* To support SCA we must migrate to using Stripe's SetupIntent API
    
## This commit:
    
* Add call to auth server to create SetupIntent
* Use secret from SetupIntent result to confirm card prior to creating a subscription
    
## Checklist

_Put an `x` in the boxes that apply_

- [ ] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-payments-server/src/lib/apiClient.ts [hunk 1]
```diff
@@ -1,5 +1,12 @@
 import { Config, defaultConfig } from './config';
-import { Plan, Profile, Customer, Subscription, Token } from '../store/types';
+import {
+  Plan,
+  Profile,
+  Customer,
+  Subscription,
+  Token,
+  FilteredSetupIntent,
```
Comment (lmorchard) [inline packages/fxa-payments-server/src/lib/apiClient.ts:8]: Might want to house this type here with the apiClient stuff. Those other types are for the Redux store, which should someday go away / [move out into fxa-shared](https://github.com/mozilla/fxa/issues/5912)

File: packages/fxa-payments-server/src/routes/ProductV2/SubscriptionCreate/index.test.tsx [hunk 2]
```diff
@@ -106,4 +111,11 @@ const defaultStripeOverride: Pick<
       error: undefined,
     };
   },
+  confirmCardSetup: async () => {
+    await wait(500);
```
Comment (lmorchard) [inline packages/fxa-payments-server/src/routes/ProductV2/SubscriptionCreate/index.test.tsx:115]: I used a delay in stories to give a change to simulate a loading spinner, but we don't need it in tests.

File: packages/fxa-payments-server/src/routes/ProductV2/SubscriptionCreate/index.tsx [hunk 3]
```diff
@@ -222,7 +230,19 @@ async function handleSubscriptionPayment({
   onRetry: (status: RetryStatus) => void;
   onSuccess: () => void;
 } & SubscriptionCreateAPIs) {
-  // 1. Create the payment method.
```
Comment (lmorchard) [inline packages/fxa-payments-server/src/routes/ProductV2/SubscriptionCreate/index.tsx]: I think this is in the wrong component - this is for creating a new subscription, where I think we want to use this SetupIntent stuff over in payment method updates in [the SubscriptionItem and PaymentUpdateForm neighborhood of components](https://github.com/mozilla/fxa/blob/main/packages/fxa-payments-server/src/routes/Subscriptions/SubscriptionItem.tsx#L75)

Comment (jackiemunroe) [reply packages/fxa-payments-server/src/routes/ProductV2/SubscriptionCreate/index.tsx]: Thanks for the clarification. Reading the docs that were linked, it seemed like we needed to use the SetupIntents for the initial payment method creation during checkout. I'll move this work 😅

**Comment by lmorchard** [comment]
Also probably want to set up [a parallel `/v2` route](https://github.com/mozilla/fxa/blob/main/packages/fxa-payments-server/src/App.tsx#L106) like I did for the subscription creation page. 

In that case, I needed enough component changes to create a new directory structure. 

But, it occurs to me that SubscriptionItem and PaymentUpdateForm components are pretty deep in the hierarchy of the Subscription route. Might be easier to add a `/v2` route that just passes in a new boolean prop to the existing Subscription route, then flip between PaymentUpdateForm (old) and a new PaymentUpdateFormV2 with the new API flow.

Maybe something like:

```
                      <Route
                        path="/v2/subscriptions"
                        exact
                        render={(props) => (
                          <SettingsLayout>
                            <Subscriptions {{ ...props, newPaymentFlow: true }} />
                          </SettingsLayout>
                        )}
                      />
```

And then plumb that into SubscriptionItem so you can render PaymentUpdateForm or PaymentUpdateFormV2 conditionally

**Comment by lmorchard** [comment]
Oh, and in case it's useful and since I'm thinking & brain dumping:

The existing PaymentUpdateForm component [uses Redux actions to manage & track API calls](https://github.com/mozilla/fxa/blob/main/packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx#L34-L36).

We can & should stop doing that. Over in the SubscriptionCreate component, I switched to [just making API calls directly](https://github.com/mozilla/fxa/blob/main/packages/fxa-payments-server/src/routes/ProductV2/SubscriptionCreate/index.tsx#L270) and manage things [like an `inProgress` state](https://github.com/mozilla/fxa/blob/main/packages/fxa-payments-server/src/routes/ProductV2/SubscriptionCreate/index.tsx#L270) locally rather than in Redux state. Way simpler.

File: packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx [hunk 4]
```diff
@@ -91,6 +125,10 @@ export const PaymentUpdateForm = ({
   const onPayment = useCallback(
     (tokenResponse: stripe.TokenResponse) => {
       if (tokenResponse && tokenResponse.token) {
+        if (useSCAPaymentFlow) {
+          // Use SetupIntents to Confirm the Card
+          handleSCA(tokenResponse.token.id);
```
Comment (lmorchard) [inline packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx:130]: FWIW, I know this is in progress, but wanted to mention: 

We're also going to want to swap in [`PaymentFormV2`](https://github.com/mozilla/fxa/blob/main/packages/fxa-payments-server/src/components/PaymentFormV2/index.tsx) over here when `useSCAPaymentFlow` is true. 

That uses the new Stripe elements library. It also has [a different `onSubmit` callback](https://github.com/mozilla/fxa/blob/main/packages/fxa-payments-server/src/routes/ProductV2/SubscriptionCreate/index.tsx#L87) that replaces `onPayment` / `onPaymentError` callbacks no longer generates a payment token.

**Comment by chenba** [comment]
There are issues with the response validators on the API side for the setupintent/create.  I see `null` for both `next_action` and `payment_method`, but the validator does not explicitly `allow(null)` on them.  So if you are getting 500s, that could be the reason.

Setting a default Payment Method in invoice settings on v1 customers with charges does appear to work.  However, the old card lingers as a payment method and the default charge for the customer.  I think we should remove the old payment methods and Payment Methods as part of the Payment Method update.

**Comment by lmorchard** [comment]
Going to close this - superseded by PR #6030

**Pull request closed without merging**